From d1cd8a21e506fd3fa9b10aa80ed4271a8e399323 Mon Sep 17 00:00:00 2001
From: Pradeep Kumar Chitrapu <quic_pradeepc@quicinc.com>
Date: Sun, 13 Feb 2022 14:30:28 -0800
Subject: [PATCH] ath12k: push HE MU-MIMO and OFDMA params to hardware for
 configuration

Currently MU-MIMO and OFDMA configurations are being set
HE IEs only. Send it to hardware as well for configuration.
Without this MU-MIMO and OFDMA is not enabled in hardware.

Signed-off-by: Pradeep Kumar Chitrapu <quic_pradeepc@quicinc.com>
Signed-off-by: Muna Sinada <quic_msinada@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -5292,7 +5292,9 @@ static void ath12k_mac_copy_he_cap(struc
 		IEEE80211_HE_MAC_CAP1_TF_MAC_PAD_DUR_MASK;
 
 	he_cap_elem->phy_cap_info[0] &=
-		~IEEE80211_HE_PHY_CAP0_CHANNEL_WIDTH_SET_160MHZ_IN_5G;
+		IEEE80211_HE_PHY_CAP0_CHANNEL_WIDTH_SET_40MHZ_IN_2G |
+		IEEE80211_HE_PHY_CAP0_CHANNEL_WIDTH_SET_40MHZ_80MHZ_IN_5G |
+		IEEE80211_HE_PHY_CAP0_CHANNEL_WIDTH_SET_160MHZ_IN_5G;
 	he_cap_elem->phy_cap_info[0] &=
 		~IEEE80211_HE_PHY_CAP0_CHANNEL_WIDTH_SET_80PLUS80_MHZ_IN_5G;
 	he_cap_elem->phy_cap_info[5] &=
